### ISSUE TITLE
Renovate is pushing test branch to master

### ISSUE THREAD

**Issue by bojovypstros** [renovatebot/renovate#18209]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

On 7. september at afternoon CEST time renovate bot hosten on GitHub started to randomly push test branch to master one. Until now it had been done 3 times and we had to stop CI/CD procedures because of this. 

### Relevant debug logs

<details><summary>Logs</summary>

[link to gist](https://gist.github.com/bojovypstros/024758648dd70220d49890f82f5622ba)

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by bojovypstros** [comment]
Link to whole job log: [here](https://gist.github.com/bojovypstros/024758648dd70220d49890f82f5622ba)

**Comment by rarkins** [comment]
Which line(s) are relevant?

**Comment by bojovypstros** [comment]
@rarkins I have no clue. Nothing seems suspicious to me about doing this thing.

**Comment by rarkins** [comment]
So how do you know it's the right log?

**Comment by bojovypstros** [comment]
Because there is no job log at the time of push to master in our repo.

**Comment by rarkins** [comment]
Here's the only commit-related information from the logs:

```
DEBUG: git commit(branch="renovate/firebase-admin-11.x")
{
  "baseBranch": "test",
  "deletedFiles": [],
  "ignoredFiles": [],
  "result": {
    "author": null,
    "branch": "renovate/firebase-admin-11.x",
    "commit": "962b773360d47da46945f5f7875cdc51429d92d2",
    "root": false,
    "summary": {
      "changes": 4,
      "insertions": 433,
      "deletions": 363
    }
  }
}

INFO: Branch created(branch="renovate/firebase-admin-11.x")
{
  "baseBranch": "test",
  "commitSha": "4225eecc79a4f27b90a3c5312b42a7a963339779"
}

DEBUG: git commit(branch="renovate/i18next-21.x")
{
  "baseBranch": "test",
  "deletedFiles": [],
  "ignoredFiles": [],
  "result": {
    "author": null,
    "branch": "renovate/i18next-21.x",
    "commit": "390ca3f7d2661e9e6e744e2ea65936f50f4ac8c8",
    "root": false,
    "summary": {
      "changes": 6,
      "insertions": 17,
      "deletions": 0
    }
  }
}

INFO: Branch created(branch="renovate/i18next-21.x")
{
  "baseBranch": "test",
  "commitSha": "d2e9e8b418bce6df9b54da4acbbdd5a5f8f0a95a"
}
```

No automerges occurred in the log you shared.

None of the above seems problematic. Until you learn more, I suggest you remove all branch automerging.

**Comment by bojovypstros** [comment]
@rarkins thanks, I saw that too. Will stop all automerging and continue monitoring the situation. Will keep you updated.

**Comment by bojovypstros** [comment]
@rarkins hello again,
after monitoring the situation for a few days with settings as they were but disabled CD since there were no problems we enabled CD again and it was okay for two weeks. But yesterday it happened again. There were 2 jobs running simultaneously as seen on screen below. I went through both of the logs and found nothing. I have uploaded them [here](https://gist.github.com/bojovypstros/ee6b8d2ba90ceca57ebc4c84317cbc29) and [here](https://gist.github.com/bojovypstros/f89fdf7cb54812a82d3cbfec47e4681e). If you have some time could you go through them try to find anything suspicious? We are really confused by this behaviour since we have been using renovate for couple years now and without any problems. Thank you!

<img width="682" alt="obrazek" src="https://user-images.githubusercontent.com/13706650/196617039-0eb703cb-b2c0-4afb-aaa7-59cd327bf1cc.png">

**Comment by rarkins** [comment]
In this log, Renovate is trying to do a branch automerge _into_ `test` branch:

```
DEBUG: Automerging branch(branch="renovate/typescript-eslint-monorepo")
{
  "baseBranch": "test"
}
DEBUG: Initializing git repository into /mnt/renovate/gh/company/company-api(branch="renovate/typescript-eslint-monorepo")
{
  "baseBranch": "test"
}
DEBUG: Performing blobless clone(branch="renovate/typescript-eslint-monorepo")
{
  "baseBranch": "test"
}
DEBUG: git clone completed(branch="renovate/typescript-eslint-monorepo")
{
  "baseBranch": "test",
  "durationMs": 4512
}
DEBUG: latest repository commit(branch="renovate/typescript-eslint-monorepo")
{
  "baseBranch": "test",
  "latestCommit": {
    "hash": "7150647ac5a7b1d435ca8e07a5bcea611bc3a1e8",
    "date": "2022-10-15T17:39:49+00:00",
    "message": "release v0.131.1",
    "refs": "HEAD -> master, tag: v0.131.1, origin/master, origin/HEAD",
    "body": "",
    "author_name": "semantic-release-bot",
    "author_email": "semantic-release-bot@martynus.net"
  }
}
INFO: Branch automerged(branch="renovate/typescript-eslint-monorepo")
{
  "baseBranch": "test"
}
```

But maybe it forgets to check out `test` before doing the merge. @zharinov can you code review this? Note that the clone isn't done until the last moment before the merge.

**Comment by bojovypstros** [comment]
As I am investigating further I see some connection between merging into test and updating version of typescript-eslint-monorepo package and problem I am describing here. The push to master was made with commit `3a186638dfc19a071b070a870b46ce6cd308fbf0` at the time of running one of the jobs (where I am sorry, I did not realize that though the time is displayed as same, there probably is some time between the two jobs 😁). The commit was exactly about updating this particular package `chore(deps): update typescript-eslint monorepo to v5.40.1`. Our renovate.json config at the time is uploaded [here](https://gist.github.com/bojovypstros/819b8116a520028ab9e1f702685c6d82) to check that there is no misconfig.
Fact I have bumped into as I am writing this comment is that the commit I was describing above is not in the test branch at all and only exists in the master branch. So the problem is not that renovate-bot is merging test branch into master as I though previously but merging one of renovate branches into master instead of the test branch and the branch is of course rebased on the test branch based right before the renovate job started.
